lxd/instance/instancetype/instance.go

```diff
@@ -1280,25 +1280,11 @@ func ConfigKeyChecker(key string, instanceType Type) (func(value string) error,
 		}
 	}
 
-	if strings.HasPrefix(key, "environment.") {
+	if shared.StringHasPrefix(key, ConfigKeyPrefixesContainer...) {
```

Comment:
we seem to have lost a check for instanceType from line 1300 here.

Reply:
I'll need to rework how `instancetype.Any` keys are validated so we don't lose this check.